akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -35,9 +57,35 @@ private[akka] object Http2Compliance {
   /** checks if the stream id was client initiated, by checking if the stream id was odd-numbered */
   final def isClientInitiatedStreamId(id: Int): Boolean = id % 2 != 0
 
+  final def requireFrameSize(size: Int, required: Int, hint: String = ""): Unit =
+    if (size != required) throw new IllegalHttp2FrameSize(size, s"MUST BE == $required. $hint")
+
   final def requireFrameSize(size: Int, max: Int): Unit =
     if (size != max) throw new IllegalHttp2FrameSize(size, s"MUST BE == $max.")
 
+  final def requireFrameSizeLessOrEqualThan(size: Int, max: Int, hint: String = ""): Unit =
+    if (size > max) throw new IllegalHttp2FrameSize(size, s"MUST BE <= $max. $hint")
+
   final def requireNoSelfDependency(id: Int, dependency: Int): Unit =
     if (id == dependency) throw new IllegalHttp2StreamDependency(id)
+
+  /** Validate value of MAX_FRAME_SIZE setting. */
+  def validateMaxFrameSize(value: Int): Unit = {
+    if (value < MinOutMaxFrameSize) throw new IllegalFrameSizeSettingException(s"MAX_FRAME_SIZE MUST NOT be < than $MinOutMaxFrameSize, attempted setting to: $value!")
+    if (value > MaxOutMaxFrameSize) throw new IllegalFrameSizeSettingException(s"MAX_FRAME_SIZE MUST NOT be > than $MaxOutMaxFrameSize, attempted setting to: $value!")
+  }
+
+  final case class WindowSizeWithOverflowProtection(value: Int) extends AnyVal {
+    def +(increment: Int): WindowSizeWithOverflowProtection = {
+      val res = value.toLong + increment
+      val resi = res.toInt
+      if (res != resi) throw new Http2Compliance.WindowOverflowException(value, increment)
```

Comment:
I actually skimmed to check what Guava does, perhaps falsely assuming it's somewhat optimal but perhaps not hm. Will move it to < checks